tuf/repository_lib.py

```diff
@@ -1520,14 +1573,22 @@ def generate_timestamp_metadata(snapshot_filename, version, expiration_date,
   tuf.formats.METADATAVERSION_SCHEMA.check_match(version)
   securesystemslib.formats.ISO8601_DATETIME_SCHEMA.check_match(expiration_date)
   securesystemslib.formats.NAME_SCHEMA.check_match(repository_name)
+  securesystemslib.formats.BOOLEAN_SCHEMA.check_match(use_length)
+  securesystemslib.formats.BOOLEAN_SCHEMA.check_match(use_hashes)
 
-  # Retrieve the versioninfo of the Snapshot metadata file.
   snapshot_fileinfo = {}
+
   length, hashes = securesystemslib.util.get_file_details(snapshot_filename,
```

Comment:
It's not the same situation as with `targets_filename` because the `snapshot_filename` value comes from here: https://github.com/theupdateframework/tuf/blob/5d40ffa3c49f6ce8ba5908abacec5630833829e8/tuf/repository_lib.py#L136
and the `fillenames` in one place have a different value than just `snapshot.json`: 
https://github.com/theupdateframework/tuf/blob/5d40ffa3c49f6ce8ba5908abacec5630833829e8/tuf/repository_tool.py#L405.

Still, I noticed an inconsistency that we use both `snapshot_filename` and the constant `SNAPSHOT_FILENAME` at the same function. Will fix that and just use `snapshot_filename`.